Bottom: 93c7700ba6929cfd5ee578192885c30bcc8ac400
Top:    c1939caea16758909da14d30532f8031973bf00c
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-03-08 18:26:53 +0000

Refresh of remove-hack-in-draw-c-which-br

---

diff --git a/src/draw.c b/src/draw.c
index e7e36df..955764c 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -895,8 +895,7 @@ DrawLayerGroup (int group, const BoxType * screen)
   Cardinal *layers = PCB->LayerGroups.Entries[group];
 
   clip_box = screen;
-  for (i = n_entries - 1; i >= 0;
-      i--, gui->set_layer (0, group, 0)) /* HACK: Subcomposite each layer in a layer group separately */
+  for (i = n_entries - 1; i >= 0; i--)
     {
       layernum = layers[i];
       Layer = PCB->Data->Layer + layers[i];
@@ -913,10 +912,6 @@ DrawLayerGroup (int group, const BoxType * screen)
 	      r_search (Layer->polygon_tree, screen, NULL, poly_callback,
 			&info);
 	      info.arg = false;
-
-	      /* HACK: Subcomposite polygons separately from other layer primitives */
-	      /* Reset the compositing */
-	      gui->set_layer (0, group, 0);
 	    }
 
 	  if (TEST_FLAG (CHECKPLANESFLAG, PCB))
diff --git a/src/hid/ps/ps.c b/src/hid/ps/ps.c
index 8a6584e..f5388af 100644
--- a/src/hid/ps/ps.c
+++ b/src/hid/ps/ps.c
@@ -652,6 +652,8 @@ ps_set_layer (const char *name, int group, int empty)
 	     && group <
 	     max_group) ? PCB->LayerGroups.Entries[group][0] : group;
 
+  printf ("set layer: %s, %i, %i (Type=%i)\n", name, group, empty, SL_TYPE (idx));
+
   if (SL_TYPE (idx) == SL_FINISHED)
     return 0;
